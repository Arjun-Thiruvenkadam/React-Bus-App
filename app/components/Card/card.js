/**
 *
 * Card
 *
 */

import styled from 'styled-components';

const CardStyle = styled.div`
  width: 25%;
  height: 100%;
  display: inline-block;
  padding: 5px 15px;
  vertical-align: top;
  
  @media (max-width: 600px) {
    width: 90%;
    height: 390px;
  }
`;

export default CardStyle;
