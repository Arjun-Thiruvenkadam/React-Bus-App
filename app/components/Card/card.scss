.Card{
  font-size: medium;
  width: 25%;
  height: 100%;
  display: inline-block;
  padding: 5px 15px;
  vertical-align: top;
}

@media(max-width:600px){
  .Card{
      width: 90%;
      height: 390px;
  }
}