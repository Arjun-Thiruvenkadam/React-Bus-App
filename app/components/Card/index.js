/**
 *
 * Card
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import './card.scss';

const Card = props => (
  <div className="Card">
    <img src={props.image} alt={props.name} />
    <h3>{props.name}</h3>
    <p>{props.children}</p>
  </div>
);

Card.propTypes = {
  image: PropTypes.any,
  name: PropTypes.string,
  children: PropTypes.any,
};

export default Card;
