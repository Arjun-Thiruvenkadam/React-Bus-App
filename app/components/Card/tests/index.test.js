import React from 'react';
import { render } from 'react-testing-library';
import Card from '../index';
import Safety from '../../../images/safety.png';

describe('<Card />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Card image={Safety} name="Safety">
        Testing...
      </Card>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
