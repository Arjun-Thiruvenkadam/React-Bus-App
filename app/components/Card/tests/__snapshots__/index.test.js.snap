// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render and match the snapshot 1`] = `
<div
  class="Card"
>
  <img
    alt="Safety"
    src="IMAGE_MOCK"
  />
  <h3>
    Safety
  </h3>
  <p>
    Testing...
  </p>
</div>
`;
